fw/procfs.c

```diff
@@ -359,13 +401,54 @@ tfw_procfs_stop(void)
 	tfw_procfs_cleanup();
 }
 
+static int
+tfw_cfgop_health_stat(TfwCfgSpec *cs, TfwCfgEntry *ce)
+{
+	int cpu;
+
+	TFW_CFG_CHECK_VAL_N(>, 0, cs, ce);
+	TFW_CFG_CHECK_NO_ATTRS(cs, ce);
+
+	health_stat_codes = kzalloc(tfw_hm_stats_size(ce->val_n),
+				    GFP_KERNEL);
+	if (!health_stat_codes)
+		return -EINVAL;
+	if (tfw_hm_stats_init_from_cfg_entry(health_stat_codes, ce)) {
+		kfree(health_stat_codes);
+		return -EINVAL;
+	}
+
+	for_each_possible_cpu(cpu) {
+		TfwPerfStat *pcp_stat = per_cpu_ptr(&tfw_perfstat, cpu);
+		pcp_stat->hm = kmalloc_node(tfw_hm_stats_size(ce->val_n),
+					    GFP_KERNEL, cpu_to_node(cpu));
+		if (!pcp_stat->hm)
+			return -EINVAL;
+		tfw_hm_stats_clone(pcp_stat->hm, health_stat_codes);
+	}
+	return 0;
+}
+
+static void
+tfw_cfgop_cleanup_health_stat(TfwCfgSpec *cs)
+{
+	int cpu;
+
+	kfree(health_stat_codes);
```

Comment:
We have a lot of problems in memory allocation/deallocation here! I use a simple garbage config
```
health_stat 1;
dsgfdsfdsfds
```
and catch kernel panic here.

If `tfw_hm_stats_init_from_cfg_entry` fails we free `health_stat_codes` but not zero this variable and have double free in
`tfw_cfgop_cleanup_health_stat` for example.

Reply:
Fixed